crates/miden-objects/src/block/account_tree.rs

```diff
@@ -128,6 +128,13 @@ impl AccountTree {
         self.smt.root()
     }
 
+    /// Returns true if the tree contains a leaf for the given account ID prefix.
+    pub fn has_account_prefix(&self, account_id: AccountId) -> bool {
+        let key = Self::id_to_smt_key(account_id);
+        let is_empty = matches!(self.smt.get_leaf(&key), SmtLeaf::Empty(_));
+        !is_empty
+    }
```

Comment:
This doesn't need to take a full `AccountId` but could just take an `AccountIdPRefix`, e.g. this also works:

```rust
pub fn account_prefix_exists(&self, account_id: AccountIdPrefix) -> bool {
    let mut key = Word::empty();
    key[Self::KEY_PREFIX_IDX] = account_id.as_felt();
    let is_empty = matches!(self.smt.get_leaf(&key), SmtLeaf::Empty(_));
    !is_empty
}
```

(If you go with this, I'd separate the key construction out into a helper function next to `id_to_smt_key`.)